### ISSUE TITLE
[BUG] Companion fails to exit on logout MacOS 14.1 (Sonoma) causing startup failure on re-login

### ISSUE THREAD

**Issue by johnhmacleod** [bitfocus/companion#2650]
### Is this a bug in companion itself or a module?

- [X] I believe this to be a bug in companion

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

[BUG] Companion fails to exit on logout MacOS 14.1 (Sonoma) causing startup failure on re-login. 
Companion: [3.1.2 (3.1.2+6243-stable-05c37a10)](https://bitfocus.io/companion/)
If it is explicitly quit from its icon, it exits just fine.
Log file shows port listen errors - port in use
Using web browser on port 8888 eventually encourages the phantom process to exit at which point Companion can be restarted.

[companion-2023-11-14.0.log](https://github.com/bitfocus/companion/files/13371371/companion-2023-11-14.0.log)

### Steps To Reproduce

1. Persuade Companion to start(!)
2. Log out user without exiting Companion from icon
3. Log in again, wait for a few seconds
4. Observe message that Companion failed to start
5. Connect browser to 127.0.0.1:8888 & observe web pages still being served!
6. Continue browsing until a "refused to connect" error is shown - now Companion has finally exited

### Expected Behavior

Companion used to exit cleanly on logout but now does not

### Environment (please complete the following information)

```markdown
- OS: MacOS 14.1
- Browser: Chrome
- Companion Version: 3.1.2 (3.1.2+6243-stable-05c37a10)
```

**Comment by johnhmacleod** [comment]
Including video of behaviour (some times have been edit out to meet <10mb reuirement!)

https://github.com/bitfocus/companion/assets/10006961/eea378bf-deb8-41eb-987a-52bd74916da2

**Comment by johnhmacleod** [comment]
It is just possible that this is caused/exacerbated by a pending connection to our ATEM (or any other Connection, I suppose) when it is offline.  ie If the ATEM is offline and the connection is in "Connecting" status, then a logout does not close Companion.  In any case, we have not seen this behaviour prior to the most recent OS X upgrade.

**Comment by Julusian** [comment]
I've found that (on linux) using alt+f4 to exit companion would have this behaviour, but not when using the quit button. I've pushed a fix for that, maybe it helps this scenario too? 
Its in the 6738 beta build

**Comment by Julusian** [comment]
I suspect this is the same issue as #2720, where the launcher window was being closed correctly, but not the companion process run inside of that. The fix from #2720 will hopefully resolve this too
